src/main/java/org/springframework/integration/kafka/core/DefaultConnectionFactory.java

```diff
@@ -172,12 +103,17 @@ public void setFetchMetadataTimeout(int fetchMetadataTimeout) {
 	 */
 	@Override
 	public BrokerAddress getLeader(Partition partition) {
-		try {
-			lock.readLock().lock();
```

Comment:
I see your point now, @artembilan - I misread your comment (for whatever reason I thought it'd referred to `getPartitions`, that uses a read lock - no reason why same pattern shouldn't be applied to `getLeaders` too.